core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -468,13 +471,13 @@ object Infer {
         case (_, Type.TyVar(b@Type.Var.Bound(_))) =>
           fail(Error.UnexpectedBound(b, t1))
         // the only vars that should appear are skolem variables, we check here
-        case (Type.TyVar(v1), Type.TyVar(v2)) if v1 == v2 => pure(())
-        case (Type.TyMeta(m1), Type.TyMeta(m2)) if m1.id == m2.id => pure(())
+        case (Type.TyVar(v1), Type.TyVar(v2)) if v1 == v2 => unit
+        case (Type.TyMeta(m1), Type.TyMeta(m2)) if m1.id == m2.id => unit
         case (Type.TyMeta(m), tpe) => unifyVar(m, tpe, r1, r2)
-        case (tpe, Type.TyMeta(m)) => unifyVar(m, tpe, r1, r2)
+        case (tpe, Type.TyMeta(m)) => unifyVar(m, tpe, r2, r1)
```

Comment:
I noticed that the right hand side switches the order of `tpe, m` from the left, but not the regions.